app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -41,6 +41,10 @@ def puppet?
   def check_override_selected
     return if (changed - ['description', 'override']).empty?
     return if override?
+    return if !override? && override_changed? && (changed - ['merge_overrides',
```

Comment:
> The first clause will always be true (since we check the opposite in the line before). 

I'm a bit torn. By explicitly checking for `!override`, line 44 makes sense in and of itself. Plus if line 43 is changed/removed, it'll still work. That said, I agree that simplifying the condition would be good so I can make this change.

> Also, If the description also changed, this check will fail although it should not

This is the current behavior on develop. I'd probably argue that this is a separate bug/issue.